File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        sorted_intervals = sorted(intervals, key = lambda interval: interval[0])
        res = []
        curr = sorted_intervals[0]
        
        for start, end in sorted_intervals[1: ]:
            if start <= curr[1]:
                curr = [curr[0], max(curr[1], end)]
            else:
                res.append(curr)
                curr = [start, end]
                
        res.append(curr)
        return res

File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& arr) {
        
        sort(arr.begin(),arr.end());
        vector<vector<int>>res;
        vector<int>a=arr[0];
        for(int i=0;i<arr.size();i++){
            if(a[1]>=arr[i][0]){
                a[1]=max(a[1],arr[i][1]);
            }
            else{
                res.push_back(a);
                a=arr[i];
            }
        }
        res.push_back(a);
        return res;
    }
};